feature: DPlayer support

Candinya opened the pull request:
## Summary

New Feature: Add [DPlayer](https://github.com/DIYgod/DPlayer) support

Videos can be played inline now, rather than opening a new window.

Related issue: #7201 

## Addition

1. ~Player language may differs (media-video.vue, Line 63, I forgot to change 😶 )~ Fixed
2. ~DPlayer full web mode would cause page style break~ ~Relieved by removing the classname immediately~ Resolved using modified version of DPlayer.

Candinya committed:
feature (test): DPlayer support

rinsuki commented on src/client/components/media-video.vue:
```diff
@@ -50,6 +50,29 @@ export default defineComponent({
+				// Only for creating new
+				this.dp = new DPlayer({
+					// Configure parameters
+					container: document.getElementById(this.video.id),
```
If multiple same video links exists in one page (e.g. renote), this conflicts with other same video link.

Candinya replied:
Reproduced. Figuring out how to resolve it. 😶

Candinya replied:
Should be fixed now. Passing `$event` into the trigger function and use `event.target` to get the clicked element. 😃

defango commented:
You may wanna look into video.js we added that and its working pretty good. https://github.com/gitmolife/gitmo/commit/02e68e896247f59a1e24da3660a56b0866c51ebf

rinsuki commented:
and I think DPlayer's danmaku feature _possibly_ violating patents of DWANGO Co., Ltd., but I'm not familiar with patents and related things, so I would like to hear from other collaborators/contributors.

Candinya committed:
fix: duplicated ids (renote) cause error

Also fixed the hide event's error, hide button style (z-index) and changed the default language and play button pointer event.

Candinya commented:
Here's one more question: DPlayer full-web (not full-screen) mode would attach a class `.dplayer-web-fullscreen-fix` containing `position: fixed;` style to body element, which will break the page style.
Figuring out how to prevent this.
A stupid method is attaching style `position: unset;`  to override when player event triggered. But there might be future problems.
Or create a special fork version ?

Candinya committed:
fix: remove attached class on full-web mode

Or just disable this mode. (Any options ?)

Candinya committed:
fix: timeline thumbnail preview error

Not the thumbnail :sweat_smile:

Candinya committed:
feat: changed to a forked version for dplayer (test)

Candinya commented:
> and I think DPlayer's danmaku feature _possibly_ violating patents of DWANGO Co., Ltd., but I'm not familiar with patents and related things, so I would like to hear from other collaborators/contributors.

I found that danmaku has been included as a W3C proposal, so there might not be legal problems. Maybe check [w3c/danmaku](https://github.com/w3c/danmaku) for more information.

acid-chicken commented:
It seems they haven't reached an opinion about this yet: w3c/danmaku#36
In my opinion, the patent aims at reducing to display overlapping of comments but W3C's approach seems quite similar to it, so (unfortunately) this player may violate their patents in Japan....

Candinya commented:
> It seems they haven't reached an opinion about this yet: [w3c/danmaku#36](https://github.com/w3c/danmaku/issues/36)
> In my opinion, the patent aims at reducing to display overlapping of comments but W3C's approach seems quite similar to it, so (unfortunately) this player may violate their patents in Japan....

Oops, so I'd close this PR to prevent further possible problems.
Thank you all for your participants and sincere help! 😃

acid-chicken commented:
Looking into this patent in more detail, I found an interesting document. According to this, the patent appears some of the claims lost their validity in 2017.
Perhaps we can avoid violating this patent by disabling following features?:
- float displayed comments
- changing text sizes per comments
- multiline comments

You can find this document by clicking "Invalidity 2017-800123" in page <https://www.j-platpat.inpit.go.jp/c1800/PU/JP-2006-333851/7294651F33633E1EBF3DEC66FAE0ECAD878D19E1829C378FC81D26BBD0A4263B/10/en>.

Candinya commented:
> Looking into this patent in more detail, I found an interesting document. According to this, the patent appears some of the claims lost their validity in ~2017~ December 2020.
> Perhaps we can avoid violating this patent by disabling following features?:
> 
> * float displayed comments
> * changing text sizes per comments
> * multiline comments
> 
> You can find this document by clicking "Invalidity 2017-800123" in page https://www.j-platpat.inpit.go.jp/c1800/PU/JP-2006-333851/7294651F33633E1EBF3DEC66FAE0ECAD878D19E1829C378FC81D26BBD0A4263B/10/en.

Yes, it's currently set as an simple player without danmaku mode ( due to some compability issues ), but if other days someone figured it out and enabled its danmaku functions, legal issues maybe occur then. 
So to prevent this possible issue, I've closed this PR and willing to try implementing another pure video player ( like video.js, @defango mentioned above )
Thanks for your sincere help 😉

acid-chicken commented:
> > According to this, the patent appears some of the claims lost their validity in 2017 December 2020.

Note: this (my deleted comment) is mistaken reading the docs. It disables the claims of disabling patents, not disabling patents. I'm sorry about that jumping to wrong conclusion.
